app/controllers/dish_photo_likes_controller.rb

```diff
@@ -5,6 +5,7 @@ class DishPhotoLikesController < ApplicationController
   def create
     if @dish_photo and !current_user.likes?(@dish_photo)
       current_user.like!(@dish_photo)
```

Comment:
Think about moving the check that a user already likes the dish to the `User#like!` method (or add a new method to `User` for this purpose). The `@dish_photo.create_activity ...` call might also benefit from being moved to `User#like!`.

Currently the controller is taking responsibility for more than it needs to about how `DishPhoto` and `likes` behave:

- it tries to stop duplicates from being created
- it creates the like activity for this

`DishPhoto` seems like the owner of these responsibilities.

"Tell, don't ask" is also relevant here, you might enjoy reviewing these:

- https://robots.thoughtbot.com/tell-dont-ask
- https://pragprog.com/articles/tell-dont-ask